saito-core/src/core/consensus/peer.rs

```diff
@@ -39,10 +40,15 @@ pub struct Peer {
     pub last_msg_at: Timestamp,
     pub wallet_version: Version,
     pub core_version: Version,
+    pub rate_limiter: RateLimiter,
 }
 
 impl Peer {
     pub fn new(peer_index: u64) -> Peer {
+        let mut rate_limiter = RateLimiter::new();
+        rate_limiter.set_limit("key_list", 10, 60_000);
+        rate_limiter.set_limit("handshake_challenge", 5, 60_000);
```

Comment:
you could remove the map use, if you create a rate limiter per these actions. (assuming we only have few of these)